server/game/costs/CostLibrary.ts

```diff
@@ -255,8 +255,8 @@ export function payPlayCardResourceCost<TContext extends AbilityContext = Abilit
 export function abilityResourceCost<TContext extends AbilityContext = AbilityContext>(amount: number | ((context: TContext) => number)): ICost<TContext> {
     return new GameSystemCost<TContext>(
         typeof amount === 'function'
-            ? GameSystems.payResourceCost<TContext>((context) => ({ target: context.player, amount: amount(context) }))
-            : GameSystems.payResourceCost<TContext>((context) => ({ target: context.player, amount }))
+            ? GameSystems.payResourceCost<TContext>((context) => ({ target: context.player, amount: amount(context), isCost: true }))
```

Comment:
I think the isCost shouldn't be needed for these the way the defaults are set up now?

Reply:
@AMMayberry1 Wouldn't they be needed now that we've removed the default true?

Reply:
Added a comment in GameSystemLibrary, ideally I think this method's default should be set to isCost: true and this shouldn't be needed. Otherwise there's no real reason to have `payResourceCost` as its own method imo